### PR TITLE
Update draft project configuration

### PR DESCRIPTION
**Pull request by kuzmavictor**
During this iteration was implemented:
- the repository layer (provided the main interface without full realization- all method return `null`);
- the necessary entities;
Was deleted the redundant entity `Avatar`. Let's store the image path.

### CODE REVIEW

File: src/main/java/io/javaclasses/whatevershop/repository/CategoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import io.javaclasses.whatevershop.entity.Category;
+
+import java.util.Collection;
+
+/**
+ * Interface for the repository pattern. Supplements the operations of the base repository with new specific
+ * operations necessary for working with categories.
+ */
+public interface CategoryRepository extends Repository<Category> {
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/CategoryRepository.java]: Who implements this interface?

File: src/main/java/io/javaclasses/whatevershop/repository/OrderRepository.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package io.javaclasses.whatevershop.repository;
+
+import io.javaclasses.whatevershop.entity.Order;
+
+public interface OrderRepository extends Repository<Order>{
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/OrderRepository.java]: Add Javadoc.

Comment (denysLavrienov) [reply src/main/java/io/javaclasses/whatevershop/repository/OrderRepository.java]: Remove empty interface or add some methods.

File: src/main/java/io/javaclasses/whatevershop/repository/ProductRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package io.javaclasses.whatevershop.repository;
+
+import io.javaclasses.whatevershop.entity.Product;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ *
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/ProductRepository.java]: Add Javadoc.

File: src/main/java/io/javaclasses/whatevershop/repository/RoleRepository.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package io.javaclasses.whatevershop.repository;
+
+import io.javaclasses.whatevershop.entity.Role;
+
+public interface RoleRepository extends Repository<Role> {
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/RoleRepository.java]: Add Javadoc.

File: src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package io.javaclasses.whatevershop.repository;
+
+import io.javaclasses.whatevershop.dto.AuthorizationToken;
+import io.javaclasses.whatevershop.entity.User;
+
+/**
+ *
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java]: Add Javadoc.

File: src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+import io.javaclasses.whatevershop.dto.AuthorizationToken;
+import io.javaclasses.whatevershop.entity.User;
+
+/**
+ *
+ */
+public interface UserRepository extends Repository<User> {
+    /**
+     *
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java]: Please refactor this Javadoc.

File: src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+     * @param login
+     * @return
+     */
+    User findUserByLogin(String login);
+    User findUserById(long id);
+    User findUserByToken(AuthorizationToken token);
+    void authUser(AuthorizationToken token, User user);
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/UserRepository.java]: `AuthorizationToken` should be returned after authentication for further usage in requests.

File: src/main/java/io/javaclasses/whatevershop/repository/ProductRepository.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+ */
+public interface ProductRepository extends Repository<Product> {
+    Collection<Product> findProductByCriteria (Map<String, Object> criteria);
```
Comment (denysLavrienov) [inline src/main/java/io/javaclasses/whatevershop/repository/ProductRepository.java]: Please refactor method argument type to `Map<String, String[]>`.

**Comment by kuzmavictor** [comment]
@denysLavrienov fixed, PTAL.

**Pull request merged**